docs: reorganize and improve documentation (#466)
scylladb-zpp-2024-javascript-driver/scylladb-javascript-driver: docs/source/statements/prepared.md

> However, not all combinations are valid — some will be rejected by the database.
> For example, SELECT statements are not permitted inside a batch.

### [Unprepared](./unprepared.md) vs Prepared

Each time a statement is executed by sending a statement string to the DB, it needs to be parsed. The driver does not parse CQL, therefore it sees statement strings as opaque.
There is an option to *prepare* a statement, i.e. parse it once by the DB and associate it with an ID. After preparation, it's enough that the driver sends the ID
and the DB already knows what operation to perform — no more expensive parsing necessary. Moreover, upon preparation the driver receives valuable data for load balancing,
enabling advanced load balancing (and better performance) of all further executions of that prepared statement.

**Key takeaway:** always use prepared statements that you are going to execute multiple times.

You can decide whether a given query will be prepared by setting the `QueryOptions.prepare` option. By default, queries are unprepared.
You do not need to manually handle statement preparation. If you enable the `prepare` option, the driver handles the rest: it either uses

> its cache of prepared statements if available, or prepares the query before execution and keeps the prepared statement in its cache from now on.

Reviewer: The original wording:

> uses its cache of prepared statements if the statement is present

was better than the new one:

> uses its cache of prepared statements if available

Author: Restored the original wording.

While it's possible to [manually provide metadata](./unprepared.md), this allows some silent bugs to sneak into user applications in case the provided metadata is invalid.

<!-- Well. While the rust driver still silently prepares with the goal of ensuring type safety. We just totally ignore the verification part. -->
:::{caution}
When an unprepared statement contains bind markers (`?`), the driver silently
prepares the statement before execution. This is especially important in batches:
for each statement with a non-empty list of values in the batch,
the driver will send a prepare request, and it will be done **sequentially**.
Results of preparation are not cached between `client.batch()` calls.

> **Do not use unprepared batches unless all statements take no bind markers.**

Reviewer: ❓ What's the point of removing helpful markers such as capitalised `NOT`?

Author: Capitalization was replaced with bold. We can restore NOT, but not in bold, especially that this is already a warning.

Author: Updated.

| INSERT, UPDATE                                 | Prepared statement if repeated, unprepared statement if once, batch if multiple statements are to be executed atomically                                             | No. While you can still use paging, it is irrelevant, because the result set of such operation is empty                                  |
| CREATE/DROP {KEYSPACE, TABLE, TYPE, INDEX,...} | Unprepared statement, batch if multiple statements are to be executed atomically                                                                                     | No. While you can still use paging, it is irrelevant, because the result set of such operation is empty                                  |

> Queries are fully asynchronous and can be run in parallel without limitation.

Reviewer: ```suggestion
Executing statements is fully asynchronous, so they can be run in parallel without limitation.
```

Author: Applied.
